common-lib/src/cmr/common/generics.clj

```diff
@@ -117,3 +119,116 @@
                        (get (json/parse-string index-raw true) :SubConceptType "X")))))
           {}
           (latest-approved-documents)))
+
+(defn read-generic-doc-file
+  "Return the specific schema's documentation files given the schema keyword name and version number.
+   if the file cannot be read, return an empty string which will have no impact on the API document.
+   Parameters:
+   * file-name: [ingest | search]
+   * generic-keyword: [:grid | ...]
+   * generic-version: 0.0.1
+   Returns: string"
+  [file-name generic-keyword generic-version]
+  (try
+    (-> "schemas/%s/v%s/%s.md"
+        (format (name generic-keyword) generic-version (name file-name))
+        (io/resource)
+        (slurp))
+    (catch Exception e (str ""))))
```

Comment:
it does look odd, just write a (debug) log stating that a schema is being skipped.

Reply:
or test if the file exists first and only act if it does, then you don't have to use a try/catch

Reply:
Added info log to